Code/GraphMol/FileParsers/cdxml_parser_catch.cpp

```diff
@@ -13,12 +13,27 @@
 #include <GraphMol/RDKitBase.h>
 #include <GraphMol/FileParsers/FileParsers.h>
 #include <GraphMol/SmilesParse/SmilesWrite.h>
+#include <GraphMol/SmilesParse/SmilesParse.h>
 #include <GraphMol/SmilesParse/SmartsWrite.h>
 #include <RDGeneral/FileParseException.h>
 #include <boost/algorithm/string.hpp>
 #include <RDGeneral/BadFileException.h>
 
 using namespace RDKit;
+std::string canon(const std::string &smi) {
+    auto *m = SmilesToMol(smi);
+    auto res = MolToSmiles(*m);
+    delete m;
+    return res;
+}
+
+void check_smiles(const RWMol &m, const std::string &expected) {
```

Comment:
maybe better to be more explicit?
```suggestion
void check_smiles_and_roundtrip(const RWMol &m, const std::string &expected) {
```